org.scala-ide.sdt.core.tests/src/org/scalaide/core/ui/completion/CompletionTests.scala

```diff
@@ -40,23 +53,42 @@ abstract class CompletionTests extends TextEditTests with CompilerSupport {
    *        proposal order
    */
   case class Completion(
-      completionToApply: String,
+      completionToApply: String = "",
       enableOverwrite: Boolean = false,
       expectedCompletions: Seq[String] = Nil,
+      expectedQualifiedCompletions: Seq[(String, String)] = Nil,
       expectedNumberOfCompletions: Int = -1,
-      respectOrderOfExpectedCompletions: Boolean = false)
+      respectOrderOfExpectedCompletions: Boolean = false,
+      sourcesToPreload: Seq[String] = Nil)
         extends Operation {
 
     override def execute() = {
-      val unit = mkScalaCompilationUnit(doc.get())
+      def basename(path: String) = {
+        path.split("/").last
+      }
 
-      val src = unit.sourceMap(doc.get.toCharArray()).sourceFile
+      def lastIndexBuilderTraversals: Seq[String] = withCompiler {
+        case indexBuilder: ScalaIndexBuilder => indexBuilder.lastIndexBuilderTraversals
+      }.get
 
-      // first, 'open' the file by telling the compiler to load it
-      unit.scalaProject.presentationCompiler { compiler =>
-        compiler.askReload(List(unit)).get
+      def loadCompilationUnit(input: String) = {
+        val unit = mkScalaCompilationUnit(input)
+        unit
+      }
 
-        compiler.askLoadedTyped(src, false).get
+      val preloadedUnits = sourcesToPreload.map(loadCompilationUnit)
+      val unit = loadCompilationUnit(source)
+      val filesToIndex = preloadedUnits.map(_.file.canonicalPath).toSet
+
+      def filesNotYetIndexed: Set[String] = {
+        filesToIndex.diff(lastIndexBuilderTraversals.toSet)
+      }
+
+      SDTTestUtils.buildWorkspace()
+
+      while(filesNotYetIndexed.nonEmpty) {
+        println(s"Waiting for ${filesNotYetIndexed.map(basename)} to be indexed (already indexed ${lastIndexBuilderTraversals.map(basename)})...")
```

Comment:
I would love to get rid of this  and the related code in `ScalaIndexBuilder`, but then I need another way to make sure that all relevant files are being indexed. Any ideas?